sdk/core/Azure.Core/aotcompatibility/test-aot-compat.ps1

```diff
@@ -0,0 +1,28 @@
+param([string]$targetNetFramework)
```

Comment:
Is there a GitHub workflow YAML file that would trigger this PowerShell script?
Check out workflow syntax for GitHub Actions [here](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions).
An example - this is how OpenTelemetry core setup AOT CI and [where](https://github.com/open-telemetry/opentelemetry-dotnet/blob/0ee80d593ce79c68936820ab2ed6f56855c44711/.github/workflows/ci-aot.yml#L32C1-L32C68)
the PowerShell script was referenced in YAML.

I would recommend to scope down this PR to just focus on bringing in AOT attributes with the current
unit test approach for faster iterations;
and use follow-up PRs for making the AOT compatibility test an CI.

Reply:
Just saw this: https://github.com/Azure/azure-sdk-for-net/pull/37734#issuecomment-1684068382. Thanks!